kia-cli/src/main/kotlin/com/faforever/ice/rpc/RpcService.kt

```diff
@@ -43,8 +44,15 @@ class RpcService(
         }
     }
 
-    fun onConnectionStateChanged(newState: String) {
-        sendNotification("onConnectionStateChanged", listOf(newState))
+    fun onConnectionStateChanged(newState: GpgnetProxy.ConnectionState) {
+        val newStateString: String? = when (newState) {
```

Comment:
Moved this logic from `GpgnetProxy` where `onConnectionStateChanged()` was called when the state changed to `CONNECTED`/`DISCONNECTED` but not `LISTENING`. Also these strings are specific to the rpc service so I think this all is better handled here than in `GpgnetProxy`